From 02cefed5ae3265e9d8c3e3bba5d38e5a5cab0b00 Mon Sep 17 00:00:00 2001
From: Bin Yang <yangbin@rock-chips.com>
Date: Mon, 16 Nov 2020 20:17:18 +0800
Subject: [PATCH] arm64: dts: rockchip: update usb configs for
 rk3566-rk817-tablet

1. set dr_mode to otg for usbdrd_dwc3.
2. remove usbhost30 and u2phy0_host, these are unused.

Signed-off-by: Bin Yang <yangbin@rock-chips.com>
Change-Id: Ia93df0660552bb54211252f4c35d72d8a1ef7934
---
 .../boot/dts/rockchip/rk3566-rk817-tablet.dts | 38 +++++++------------
 1 file changed, 13 insertions(+), 25 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
index 5aeea3e3197d..81b6ad857243 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
@@ -383,10 +383,6 @@
 	status = "okay";
 };
 
-&hdmi_sound {
-	status = "okay";
-};
-
 &hdmi_in_vp0 {
 	status = "okay";
 };
@@ -395,6 +391,10 @@
 	status = "disabled";
 };
 
+&hdmi_sound {
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
 
@@ -897,7 +897,6 @@
 	status = "okay";
 };
 
-
 &sdmmc0 {
 	max-frequency = <150000000>;
 	supports-sd;
@@ -939,34 +938,23 @@
 	pinctrl-0 = <&uart1m0_xfer &uart1ctsnm0>;
 };
 
-&usb2phy0 {
+&u2phy0_otg {
 	status = "okay";
-
-	u2phy0_host: host-port {
-		status = "okay";
-	};
-
-	u2phy0_otg: otg-port {
-		status = "okay";
-	};
 };
 
-&usbdrd30 {
+&usb2phy0 {
 	status = "okay";
-
-	usbdrd_dwc3: dwc3@fcc00000 {
-		dr_mode = "peripheral";
-		maximum-speed = "high-speed";
-		status = "okay";
-	};
 };
 
-&usbhost30 {
+&usbdrd_dwc3 {
+	dr_mode = "otg";
+	extcon = <&usb2phy0>;
+	maximum-speed = "high-speed";
 	status = "okay";
+};
 
-	usbhost_dwc3: dwc3@fd000000 {
-		status = "okay";
-	};
+&usbdrd30 {
+	status = "okay";
 };
 
 &vdpu {
-- 
2.35.3

